calibration_guis/src/main_window.cpp

```diff
@@ -0,0 +1,206 @@
+  msgBox.setText("Couldn't find the ros master.");
+  msgBox.exec();
+    close();
+}
+/**
+ * This function is signalled by the underlying model. When the model changes,
+ * this will drop the cursor down to the last line in the QListview to ensure
+ * the user can always see the latest log message.
+ */
+void MainWindow::updateLoggingView() {
+}
+
+/*****************************************************************************
+** Implementation [Menu]
+*****************************************************************************/
+
+void MainWindow::on_actionAbout_triggered() {
+    QMessageBox::about(this, tr("About ..."),tr("<h2>PACKAGE_NAME Test Program 0.10</h2><p>Copyright Yujin Robot</p><p>This package needs an about description.</p>"));
+}
+
+/*****************************************************************************
+** Implementation [Configuration]
+*****************************************************************************/
+
+void MainWindow::ReadSettings() {
+    QSettings settings("Qt-Ros Package", "calibration_guis");
+    restoreGeometry(settings.value("geometry").toByteArray());
+    restoreState(settings.value("windowState").toByteArray());
+    QString master_url = settings.value("master_url",QString("http://192.168.1.2:11311/")).toString();
+    QString host_url = settings.value("host_url", QString("192.168.1.3")).toString();
+    bool remember = settings.value("remember_settings", false).toBool();
+    bool checked = settings.value("use_environment_variables", false).toBool();
+    if ( checked ) {
+    }
+}
+
+void MainWindow::WriteSettings() {
+    QSettings settings("Qt-Ros Package", "calibration_guis");
+    settings.setValue("geometry", saveGeometry());
+    settings.setValue("windowState", saveState());
+
+
+}
+
+void MainWindow::closeEvent(QCloseEvent *event)
+{
+  WriteSettings();
+  QMainWindow::closeEvent(event);
+}
+
+void calibration_guis::MainWindow::on_startButton_clicked()
+{
+  std_srvs::Trigger srv;
+  if (start_client1_.call(srv))
+  {
+    ROS_INFO("Output %s",srv.response.message.c_str());
+  }
+  else
+  {
+    ROS_ERROR("Failed to call service Trigger");
+  }
+
+}
+
+void calibration_guis::MainWindow::on_obsButton_clicked()
+{
+  std_srvs::Trigger srv;
+  if (obs_client2_.call(srv))
+  {
+    ROS_INFO("Output %s",srv.response.message.c_str());
+  }
+  else
+  {
+    ROS_ERROR("Failed to call service Trigger");
+  }
+}
+
+void calibration_guis::MainWindow::on_saveButton_clicked()
+{
+  std_srvs::Trigger srv;
+  if (save_client4_.call(srv))
+  {
+    ROS_INFO("Output %s",srv.response.message.c_str());
+  }
+  else
+  {
+    ROS_ERROR("Failed to call service Trigger");
+  }
+}
+
+void calibration_guis::MainWindow::on_covButton_clicked()
+{
+  std_srvs::Trigger srv;
+  if (cov_client5_.call(srv))
+  {
+    ROS_INFO("Output %s",srv.response.message.c_str());
+  }
+  else
+  {
+    ROS_ERROR("Failed to call service Trigger");
+  }
+}
+
+void calibration_guis::MainWindow::on_runButton_clicked()
+{
```

Comment:
add a text box where one enters the allowable cost for the run

Reply:
I did it is the accessed from the line: " std::string allowed_residual_string = ui.allowed_residual_->text().toStdString(); "